elytron/ELY-1921-http-external-mechanism.adoc

```diff
@@ -0,0 +1,71 @@
+= HTTP External Mechanism Support in Elytron
+:author:            Ashley Abdel-Sayed
+:email:             aabdelsa@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+This proposal is to add support for the External HTTP mechanism with Elytron. This mechanism will perform no verification,
+but will authenticate a user based on credentials established externally. This will be used to accept a principal which
+is passed by the `REMOTE_USER` attribute in the AJP protocol.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/ELY-1921[ELY-1921]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1323[EAP7-1323]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE.
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+* Security
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* New HTTP `ExternalAuthenticationMechanism`, `ExternalMechanismFactory`, and `WildFlyElytronHttpExternalProvider` classes
+* Ability to pass `REMOTE_USER` attribute from undertow to Elytron (via elytron-web)
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+== Test Plan
+
+* Tests will be added to the https://github.com/wildfly-security/wildfly-elytron/tree/1.x/tests[wildfly-elytron testsuite] to test the new mechanism's functionality
+* Tests will be added to the https://github.com/wildfly-security/elytron-web/tree/1.x/undertow/src/test[elytron-web/undertow testsuite] to test successful HTTP authentication with the External
+mechanism
+
+== Community Documentation
+
+* Documentation will be added to https://github.com/wildfly/wildfly/blob/master/docs/src/main/asciidoc/_elytron/Using_the_Elytron_Subsystem.adoc[Using the Elytron Subsystem]
+describing how to configure authentication from an external source
+* Documentation will be added to https://github.com/wildfly/wildfly/blob/master/docs/src/main/asciidoc/_high-availability/subsystem-support/mod_cluster_Subsystem.adoc[mod_cluster Subsystem] describing how to configure
+Elytron to authenticate a `REMOTE_USER` via the External HTTP Mechanism
+
+== Release Note Content
+Users are now able to configure Elytron to use credentials established externally from the server to authenticate the
```

Comment:
Just a very minor comment - we should mention here that the new support is for HTTP since we do already support the EXTERNAL mechanism with SASL.

Reply:
Updated, thanks!